src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -101,6 +101,35 @@ public override bool Execute()
             return true;
         }
 
+        // Filenames with special characters like '#' and'{' get written
+        // into the section names in the resulting .editorconfig file. Later,
+        // when the file is parsed in configuration options these special
+        // characters are interpretted as invalid values and ignored by the
+        // processor. We encode the special characters in these strings
+        // before writing them here.
+
+        private static string EncodeString(string p)
+        {
+            StringBuilder builder = new StringBuilder();
+            for (var i = 0; i < p.Length; i++)
```

Comment:
`i` is only used for indexing, so thsi can just be: `foreach (var c in p) ... c switch ...`